wrangler-service/src/main/java/co/cask/wrangler/service/s3/S3Service.java

```diff
@@ -239,18 +241,26 @@ public void run(DatasetContext datasetContext) throws Exception {
       listObjectsRequest.setDelimiter("/");
       ObjectListing result;
       DirectoryListing listing = new DirectoryListing();
+      boolean limitExceeded = false;
       do {
+        if (listing.size() == BUCKET_LIMIT) {
```

Comment:
I think it's better to make this a query param with a default value instead of hardcoded. It's not much more work and offers a possibility of the UI tweaking it if needed.

Reply:
Yea I agree.